### ISSUE TITLE
Relayer clients initialization cleanup

### ISSUE THREAD

**Issue by ancazamfir** [informalsystems/ibc-rs#53]
After restarting the relayer that had previously sync-ed some header, there are some debug messages about hashes not matching and trusted options height < header height in store. I think we need to look into this a bit, not sure why they occur as this is seen with good providers/ chains. 
(see TODO logs below)
```
    Finished dev [unoptimized + debuginfo] target(s) in 0.39s
     Running `target/debug/relayer -v -c simple_config.toml start`
Apr 12 13:58:03.207 DEBUG abscissa_core::component::registry: registered component: abscissa_core::terminal::component::Terminal (v0.5.2)
Apr 12 13:58:03.208 DEBUG abscissa_core::component::registry: registered component: abscissa_core::trace::component::Tracing (v0.5.2)
Apr 12 13:58:03.208 DEBUG abscissa_core::component::registry: registered component: abscissa_tokio::TokioComponent (v0.5.1)
Apr 12 13:58:03.209 DEBUG relayer_cli::commands::start: launching 'start' command
Apr 12 13:58:03.209  INFO relayer_cli::commands::start: spawning light client chain.id=ibc1
Apr 12 13:58:03.210  INFO relayer_cli::commands::start: spawning light client chain.id=ibc0
Apr 12 13:58:03.210  INFO relayer_cli::commands::start: starting relayer
Apr 12 13:58:03.210  INFO relayer_cli::relayer: relayer is running
     Relayer spawning light client for chain ibc1
Apr 12 13:58:03.373 DEBUG relayer::client: initializing client from trusted store chain.id=ibc1
Apr 12 13:58:03.373 DEBUG relayer::client: restoring trusted state from store chain.id=ibc1
Apr 12 13:58:03.374 DEBUG relayer::client: found trusted state in store chain.id=ibc1 last_height=5321
Apr 12 13:58:03.374 DEBUG relayer::client: found last trusted state
Apr 12 13:58:03.374 DEBUG relayer::client: restoring trusted state from store chain.id=ibc1
Apr 12 13:58:03.374 DEBUG relayer::client: trusted options height is lesser than header height in trust store. TODO: rollback chain.id=ibc1 trust_options.height=806 trusted_header.height=5321
Apr 12 13:58:03.375 DEBUG relayer::client: comparing trusted_header.hash against primary.hash chain.id=ibc1 primary.hash=F24C779098C80571ACD8C399D4CBDC60DD213BC1A51546D143FCD238037713E2 trusted_header.hash=B6C7EBBC34C558CD4DD1DDA3F69F79B86CFC59813F3576DB8CCAF6B26BADFAFF
Apr 12 13:58:03.376 DEBUG relayer::client: hash do not match! TODO: cleanup chain.id=ibc1 primary.hash=F24C779098C80571ACD8C399D4CBDC60DD213BC1A51546D143FCD238037713E2 trusted_header.hash=B6C7EBBC34C558CD4DD1DDA3F69F79B86CFC59813F3576DB8CCAF6B26BADFAFF
Apr 12 13:58:03.376 DEBUG relayer::client: headers hashes match chain.id=ibc1 primary.hash=F24C779098C80571ACD8C399D4CBDC60DD213BC1A51546D143FCD238037713E2 trusted_header.hash=B6C7EBBC34C558CD4DD1DDA3F69F79B86CFC59813F3576DB8CCAF6B26BADFAFF
Apr 12 13:58:03.376 DEBUG relayer::client: trusted header was valid
     Relayer spawning light client for chain ibc0
```

**Comment by romac** [comment]
This occurs because we do not implement rollback yet: https://github.com/informalsystems/ibc-rs/blob/287bd64409b58f886a9d48c0108249d6fff1f9ac/relayer/relay/src/client.rs#L346

As per the doc comments in the Go relayer:

> //     3) behind trustedHeader.Height => remove all the headers between
> //     options.Height and trustedHeader.Height, update trustedHeader, then
> //     check options.Hash matches the trustedHeader's hash (if not, remove
> //     trustedHeader and all the headers before)
> 
> // The intuition here is the user is always right. I.e. if she decides to reset
> // the light client with an older header, there must be a reason for it.

**Comment by ancazamfir** [comment]
We need to allow the relayer (re)start from latest store state. Currently it always starts from the trust options in the store. So probably allow parameters to the start command.

This is done here in the Go code:
https://github.com/tendermint/tendermint/blob/fb35b474ad9321624ba06be135825a469b0ea684/cmd/tendermint/commands/lite.go#L109-L130
caller docs:
https://github.com/tendermint/tendermint/blob/fb35b474ad9321624ba06be135825a469b0ea684/cmd/tendermint/commands/lite.go#L37-L45

**Comment by ancazamfir** [comment]
Fixed in #54
